internal/jobservice/repository/postgres.go

```diff
@@ -245,11 +249,15 @@ func (s *JSRepoPostgres) SubscribeJobSet(ctx context.Context, queue string, jobS
 
 // UnSubscribe to JobSet and delete all the jobs in the database
 func (s *JSRepoPostgres) CleanupJobSetAndJobs(ctx context.Context, queue string, jobSet string) (int64, error) {
```

Comment:
with the new foreign-key constraint, we should delete the jobs first before the jobset. Also, the function is expected to return the number of affected jobs

Reply:
could we cascade delete from jobset?

Reply:
Can you have a look at it now ? I am thinking also of deleting `CleanupJobSetAndJobs` from the `SQLJobService` interface since it is just calls `UnsubscribeJobSet` in both of it's implemetation to perform the deletion due to cascade delete.